libgobuster/dir.go

```diff
@@ -159,7 +148,7 @@ func ProcessDirEntry(s *State, word string, resultChan chan<- Result) {
 	// TODO: Error propagation handling
 	dirResp, dirSize, err := GoGet(s, s.Url, prefix+word+suffix, s.Cookies)
 	if err != nil {
-		panic(err)
+		//panic(err)
```

Comment:
I made this terrible thing here because I realized that the 302 redirect was causing a panic here.

Reply:
Which '302 redirect'? Can you give a specific example/site it happens on for debugging.

Definitely will need to actually do something with the error here before it can be merged, otherwise it will fail silently, and that's bad.

Without running it/seeing it in practice I can't really give a good idea of how it should be handled. If it's returning an error where it isn't really an error, then we can probably catch that case and handle it somewhere appropriately.

Reply:
Funny thing is that now, when trying to reproduce it, I realised that it's happening in the host I was testing it, for a 302 redirect:

```./gobuster -u https://king.com/ -w <(echo -n "images")

Gobuster v1.4.1              OJ Reeves (@TheColonial)
=====================================================
=====================================================
[+] Mode         : dir
[+] Url/Domain   : https://king.com/
[+] Threads      : 10
[+] Wordlist     : /dev/fd/63
[+] Status codes : 200,204,301,302,307
=====================================================
panic: Redirect code: 302

goroutine 15 [running]:
panic(0x22a900, 0xc4203d1ea8)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/bberastegui/gobuster/libgobuster.ProcessDirEntry(0xc4200ce000, 0xc4203dc16a, 0x6, 0xc4203109c0)
	/[REDACTED]/go/src/github.com/bberastegui/gobuster/libgobuster/dir.go:151 +0x438
github.com/bberastegui/gobuster/libgobuster.Process.func1(0xc420310960, 0xc4200ce000, 0xc4203109c0, 0xc4203dc1c0)
	/[REDACTED]/go/src/github.com/bberastegui/gobuster/libgobuster/state.go:116 +0x8f
created by github.com/bberastegui/gobuster/libgobuster.Process
	/[REDACTED]/go/src/github.com/bberastegui/gobuster/libgobuster/state.go:122 +0x1a9```

But not on other redirects like 301, for example.